src/Plugins/ComplexCore/src/ComplexCore/Filters/InterpolatePointCloudToRegularGridFilter.cpp

```diff
@@ -68,36 +69,45 @@ void mapPointCloudDataByKernel(IDataArray* source, INeighborList* dynamic, std::
 
 void mapPointCloudDataByKernel(IDataArray* source, INeighborList* dynamic, std::vector<float>& kernelVals, int64 kernel[3], usize dims[3], usize curX, usize curY, usize curZ, usize vertIdx)
 {
-  switch(source->getDataType())
+  // Because we are using INeighborList we CANNOT accept Boolean template parameters. So we convert
+  // to a NumericType from DataType and switch on that. This will stop the compiler warning but
+  // silently skip over the boolean types. Not really sure if this is a problem.
+  auto numericTypeResult = ConvertDataTypeToNumericType(source->getDataType());
+  if(!numericTypeResult.has_value())
   {
-  case DataType::int8:
+    return;
+  }
+
+  switch(numericTypeResult.value())
```

Comment:
Instead of converting, could be add a case for `DataType::boolean` that just returns?

Reply:
We are converting to NumericType which does not have boolean as a valid type. I can certainly just check for the DataType == boolean and then return, but that is what effectively the code does:
```  
auto numericTypeResult = ConvertDataTypeToNumericType(source->getDataType());
  if(!numericTypeResult.has_value()) // This will be true if the DataType == boolean.
  {
    return;
  }
```

Reply:
Functionally we end up skipping bool arrays either way, but a single additional case seems simpler to me. We have handle all the values in `DataType` regardless so going through `NumericType` doesn't gain much.

Reply:
```
  case DataType::boolean:
    break;
  }
```

Reply:
Done